api/app/controllers/spree/api/images_controller.rb

```diff
@@ -15,7 +15,11 @@ def show
 
       def create
         authorize! :create, Image
-        @image = scope.images.create(image_params)
+        if valid_url? image_params[:attachment]
+          @image = scope.images.create(attachment: URI.open(image_params[:attachment]))
```

Comment:
Looking twice, is this removing the other attributes in `image_params` and creating an image with only the attachment? I think we should update the `image_params` instead, preserving the rest of the attributes.

Reply:
I think it may be.  I'll do some debugging soon and get back to you.

Reply:
@kennyadsl  Good catch. I added a fix that addresses your concern.  The line following the line you commented on now updates the resultant `Spree::Image` object with the other  attributes within `image_params`, except for `:attachment`

I tried updateing `image_params` instead, as per your suggestion, but that resulted in the following error when running the spec: 

```
Paperclip::AdapterRegistry::NoHandlerError:
       No handler found for "https://github.com/solidusio/brand/raw/1827e7afb7ebcf5a1fc9cf7bf6cf9d277183ef11/PNG/solidus-logo-dark.png"

```

So, I opted to create the image first using `image_params[:attachment]`, and then update all the other attributes using `ActionController::Parameters` `except` method.

I also added a test that fails without the newly added line

Reply:
It sounds weird... I'll try to take a look soon and see if I find why this happens. I'd prefer if we can make a single `INSERT` query here, instead of also updating the record later.

Reply:
I'm trying locally and I got what you mean now. It looks like a bug in paperclip: if it doesn't have the adapter with `create` and all the rest of attributes I can't get why it has the adapter if you create the record only with the attachment one...

Anyway, I think we can avoid the double query with this code:

```ruby
# frozen_string_literal: true

module Spree
  module Api
    class ImagesController < Spree::Api::BaseController
      def create
        authorize! :create, Image

        @image = scope.images.build(image_params.except(:attachment))
        @image.attachment = prepared_attachment
        @image.save

        respond_with(@image, status: 201, default_template: :show)
      end

      private

      def prepared_attachment
        uri = URI.parse image_params[:attachment]

        if uri.is_a? URI::HTTP
          URI.open(image_params[:attachment])
        else
          image_params[:attachment]
        end
      rescue URI::InvalidURIError
        image_params[:attachment]
      end
    end
  end
end
```

I'm also wondering if we need something similar in the update method.

Reply:
@calebhaye I was wondering if you had a chance to look at this? Would be really cool to get this merged. 🙂

Reply:
@aldesantis yep!  Just pushed the latest, which includes an updated `update` method and a new spec for testing the update method when updating the attachment property.

@kennyadsl please note: I originally updated the `update` method to look like this:

```
@image.assign_attributes(image_params.except(:attachment))
@image.attachment = prepared_attachment if image_params[:attachment].present?
@image.save
```
But I changed it to the following because I feel as though it reads more logically:  
```
if image_params[:attachment].present?
  @image.assign_attributes(image_params.except(:attachment))
  @image.attachment = prepared_attachment
  @image.save
else
  @image.update image_params
end
```
I'd rather have it make more sense to developers who reference the code in the future, than to have fewer lines